packages/core/src/services/sessionSummaryService.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type { MessageRecord } from './chatRecordingService.js';
+import type { BaseLlmClient } from '../core/baseLlmClient.js';
+import { partListUnionToString } from '../core/geminiRequest.js';
+import { debugLogger } from '../utils/debugLogger.js';
+import type { Content } from '@google/genai';
+import { getResponseText } from '../utils/partUtils.js';
+
+const DEFAULT_MAX_MESSAGES = 20;
+const DEFAULT_MAX_CHARS = 80;
+const DEFAULT_TIMEOUT_MS = 5000;
+
+const SUMMARY_PROMPT = `Summarize the user's primary intent or goal in this conversation in ONE sentence (max 80 characters).
+Focus on what the user was trying to accomplish, not the technical details.
```

Comment:
decided to keep details out but maybe we want technical details?

Reply:
I'd omit the part about "not the technical details" unless experimentation shows we really need it for the results to look good.
Could imagine for more technical propmts I do I wouldn't want them to be overly simplified.